src/Backend/Modules/Authentication/Tests/Actions/IndexTest.php

```diff
@@ -77,5 +92,77 @@ public function testAuthenticationWithCorrectCredentials()
             'now editing:',
             $client->getResponse()->getContent()
         );
+
+        // logout to get rid of this session
+        $client->followRedirects(false);
+        $client->request('GET', '/private/en/authentication/logout');
+    }
+
+    /**
+     * Login as a pages user.
+     * This user has the rights to access only the pages module.
+     */
+    public function testPagesUserWithCorrectCredentials()
+    {
+        $client = static::createClient();
+        $client->followRedirects();
+        $client->setMaxRedirects(10);
+
+        $crawler = $client->request('GET', '/private/en/authentication');
+        $this->assertEquals(
+            200,
+            $client->getResponse()->getStatusCode()
+        );
+
+        $form = $crawler->selectButton('login')->form();
+        $this->submitForm($client, $form, array(
+            'form' => 'authenticationIndex',
+            'backend_email' => 'pages-user@fork-cms.com',
+            'backend_password' => 'fork',
+            'form_token' => $form['form_token']->getValue(),
+        ));
+
+        $this->assertContains(
+            'Recently edited',
+            $client->getResponse()->getContent()
+        );
+
+        // logout to get rid of this session
+        $client->followRedirects(false);
+        $client->request('GET', '/private/en/authentication/logout');
+    }
+
+    /**
+     * Login as a users user.
+     * This user only has the rights to access the users edit action.
+     * It should enable the user to edit his own user-account.
+     */
+    public function testUsersUserWithCorrectCredentials()
+    {
+        $client = static::createClient();
+        $client->setMaxRedirects(2);
```

Comment:
2 redirects should be enough:
* user logs in and gets redirected to the first allowed action (being Users Edit);
* Users Edit can't find an id parameter and redirects to itself with the current user id.

In fact, 1 redirect should be enough for the testPagesUserWithCorrectCredentials test.

Setting the max redirects as tight as can be could help rule out unexpected behavior.